src/main/java/mil/dds/anet/resources/EntityAvatarResource.java

```diff
@@ -0,0 +1,116 @@
+package mil.dds.anet.resources;
+
+import io.leangen.graphql.annotations.GraphQLArgument;
+import io.leangen.graphql.annotations.GraphQLMutation;
+import io.leangen.graphql.annotations.GraphQLQuery;
+import io.leangen.graphql.annotations.GraphQLRootContext;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.Response;
+import java.util.Map;
+import java.util.Optional;
+import mil.dds.anet.AnetObjectEngine;
+import mil.dds.anet.beans.EntityAvatar;
+import mil.dds.anet.beans.Person;
+import mil.dds.anet.database.EntityAvatarDao;
+import mil.dds.anet.database.OrganizationDao;
+import mil.dds.anet.utils.AnetAuditLogger;
+import mil.dds.anet.utils.AuthUtils;
+import mil.dds.anet.utils.DaoUtils;
+
+public class EntityAvatarResource {
+
+  private final EntityAvatarDao entityAvatarDao;
+
+  public EntityAvatarResource(AnetObjectEngine engine) {
+    this.entityAvatarDao = engine.getEntityAvatarDao();
+  }
+
+  /**
+   * Gets the avatar for this relatedObject uuid, null if none existing
+   *
+   * @param relatedObjectUuid the relatedObjectUuid
+   * @return the entity avatar
+   */
+  @GraphQLQuery(name = "entityAvatar")
+  public EntityAvatar getEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "relatedObjectType") String relatedObjectType,
+      @GraphQLArgument(name = "relatedObjectUuid") String relatedObjectUuid) {
+    return entityAvatarDao.getByRelatedObject(relatedObjectType, relatedObjectUuid).orElse(null);
+  }
+
+  /**
+   * Creates or updates an entity avatar
+   * 
+   * @param context the context
+   * @param entityAvatar the entity avatar
+   * @return the entity avatar
+   */
+  @GraphQLMutation(name = "createOrUpdateEntityAvatar")
+  public Integer createOrUpdateEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "entityAvatar") EntityAvatar entityAvatar) {
+    final int numRows;
+    final Person user = DaoUtils.getUserFromContext(context);
+
+    // Check if user is authorized to manipulate avatars for this relatedObjectUuid
+    assertPermission(user, entityAvatar.getRelatedObjectType(),
+        entityAvatar.getRelatedObjectUuid());
+
+    // Do we have an avatar already for this entity?
+    final Optional<EntityAvatar> dbEntityAvatar = entityAvatarDao.getByRelatedObject(
+        entityAvatar.getRelatedObjectType(), entityAvatar.getRelatedObjectUuid());
+
+    if (dbEntityAvatar.isEmpty()) {
+      numRows = entityAvatarDao.insert(entityAvatar);
+      AnetAuditLogger.log("Avatar for entity {} created by {}", entityAvatar.getRelatedObjectUuid(),
+          user);
+    } else {
+      numRows = entityAvatarDao.update(entityAvatar);
+      AnetAuditLogger.log("Avatar for entity {} updated by {}", entityAvatar.getRelatedObjectUuid(),
+          user);
+    }
```

Comment:
Please also log the entity type (prevents having to guess which table was updated).